MobileCenter/MobileCenter/Internals/Channel/MSChannelDefault.m

```diff
@@ -392,4 +399,56 @@ - (void)notifyFailureBeforeSendingForItem:(id<MSLog>)item withError:(NSError *)e
   }
 }
 
+#pragma mark – Observers
+
+- (void)addObservers {
+
+// There is no need to do trigger sending on macOS because we can just continue to execute tasks there.
+#if !TARGET_OS_OSX
+  if (!MS_IS_APP_EXTENSION) {
+    __weak typeof(self) weakSelf = self;
+    if (self.appDidEnterBackgroundObserver == nil) {
+      void (^notificationBlock)(NSNotification *note) = ^(NSNotification __unused *note) {
```

Comment:
I think this will run in the observed object thread, it might leads to race conditions. Should make any unprotected call to the channel thread safe `flushQueue` can remain asynchronous but `suspend` in the expiration handler will probably need to be synchronous yet in the channel queue.